### PR TITLE
Move to BusIO

### PR DESCRIPTION
**Pull request by caternuson**
This is for #10.

**BREAKING**
The ctors and begin() calls have been changed slightly to be more in line with other libraries. addr has been moved to begin() and the parameter to set gyro range has been removed. Example still works as expected:
Screenshot from 2021-06-23 10-50-19

### CODE REVIEW

**Comment by ladyada** [comment]
this is gtg. note the sensor itself is no longer manufactured. do you wanna check in before doing busio ports to get more popular/active chips done first?

File: Adafruit_FXAS21002C.cpp [hunk 1]
```diff
@@ -336,16 +217,13 @@ void Adafruit_FXAS21002C::getSensor(sensor_t *sensor) {
 */
 /**************************************************************************/
 void Adafruit_FXAS21002C::standby(boolean standby) {
+  Adafruit_BusIO_Register CTRL_REG1(i2c_dev, GYRO_REGISTER_CTRL_REG1);
+  Adafruit_BusIO_RegisterBits active_bit(&CTRL_REG1, 1, 1);
 
-  uint8_t reg1 = read8(GYRO_REGISTER_CTRL_REG1);
   if (standby) {
-    reg1 &= ~(0x03);
```
Comment (ladyada) [inline Adafruit_FXAS21002C.cpp:342]: looks like it sets/unsets the bottom two bits?

File: Adafruit_FXAS21002C.cpp [hunk 2]
```diff
@@ -336,16 +217,13 @@ void Adafruit_FXAS21002C::getSensor(sensor_t *sensor) {
 */
 /**************************************************************************/
 void Adafruit_FXAS21002C::standby(boolean standby) {
+  Adafruit_BusIO_Register CTRL_REG1(i2c_dev, GYRO_REGISTER_CTRL_REG1);
+  Adafruit_BusIO_RegisterBits active_bit(&CTRL_REG1, 1, 1);
```
Comment (ladyada) [inline Adafruit_FXAS21002C.cpp]: this is for a single bit

File: Adafruit_FXAS21002C.cpp [hunk 3]
```diff
@@ -120,91 +92,23 @@ Adafruit_FXAS21002C::Adafruit_FXAS21002C(int32_t sensorID, byte addr) {
...
-  _range = rng;
-
-  /* Clear the raw sensor data */
-  raw.x = 0;
-  raw.y = 0;
-  raw.z = 0;
-
-  /* Make sure we have the correct chip ID since this checks
-     for correct address and that the IC is properly connected */
-  uint8_t id = read8(GYRO_REGISTER_WHO_AM_I);
```
Comment (ladyada) [inline Adafruit_FXAS21002C.cpp:143]: we should still check the ID register too

**Review by ladyada — APPROVED** [review]

File: Adafruit_FXAS21002C.cpp (hunk 2 above)
Comment (caternuson) [reply Adafruit_FXAS21002C.cpp]: thanks. good catch.

File: Adafruit_FXAS21002C.cpp (hunk 3 above)
Comment (caternuson) [reply Adafruit_FXAS21002C.cpp:143]: yep. thats done in `begin()`.

Comment (ladyada) [reply Adafruit_FXAS21002C.cpp:143]: oh duh

**Review by ladyada — APPROVED** [review]

**Comment by ladyada** [comment]
kk can merge!

**Comment by caternuson** [comment]
>do you wanna check in before doing busio ports to get more popular/active chips done first?

sure. these two (also PRd for FXOS8700) were a couple of issues i had opened, so were on my notional list / radar.

**Pull request merged**